### PR TITLE
Speedreader bubble graphic

### PR DESCRIPTION
**Pull request by keur**
Resolves https://github.com/brave/brave-browser/issues/17044

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by keur** [comment]
@karenkliu  Attached pictures of both states, and dark theme and light theme.

Screenshot_20210719_172544

Screenshot_20210719_172404

File: browser/ui/views/speedreader/speedreader_dancing_books.cc [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    SkColorSetRGB(0x49, 0x50, 0x57);
+
+constexpr int kMinimumWidth = 287;
+constexpr int kMinimumHeight = 61;
+
+}  // anonymous namespace
+
+namespace speedreader {
+
+SpeedreaderDancingBooks::SpeedreaderDancingBooks() : views::View() {
```
Comment (simonhong) [inline browser/ui/views/speedreader/speedreader_dancing_books.cc:28]: don't need to call `View()` explicitely.

File: browser/ui/views/speedreader/speedreader_dancing_books.h [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+#include <utility>
+
+#include "ui/gfx/geometry/vector2d.h"
+#include "ui/native_theme/themed_vector_icon.h"
+#include "ui/views/view.h"
+
+namespace speedreader {
+
+using BookGraphic = std::pair<gfx::Vector2d, const gfx::VectorIcon*>;
```
[the remark this answers is not in the document]
Comment (simonhong) [reply browser/ui/views/speedreader/speedreader_dancing_books.h:17]: And, how about making `graphics_locations_` as static const vars?

[the remark this answers is not in the document]
Comment (keur) [reply browser/ui/views/speedreader/speedreader_dancing_books.h]: good idea. fixed

File: browser/ui/views/speedreader/reader_mode_bubble.cc [hunk 3]
```diff
@@ -128,6 +129,9 @@ void ReaderModeBubble::Init() {
       l10n_util::GetStringUTF16(IDS_SPEEDREADER_ENABLE_BUTTON));
   enable_speedreader_button_ =
       AddChildView(std::move(enable_speedreader_button));
+
+  auto speedreader_dancing_books = std::make_unique<SpeedreaderDancingBooks>();
+  AddChildView(std::move(speedreader_dancing_books));
```
[the remark this answers is not in the document]
Comment (keur) [reply browser/ui/views/speedreader/reader_mode_bubble.cc]: done

File: browser/ui/views/speedreader/speedreader_dancing_books.cc [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+void SpeedreaderDancingBooks::OnPaint(gfx::Canvas* canvas) {
+  SkColor color =
+      ui::NativeTheme::GetInstanceForNativeUi()->ShouldUseDarkColors()
+          ? kBraveSpeedreaderGraphicDark
+          : kBraveSpeedreaderGraphicLight;
+
+  int xclip = 0;
+  int clip_width = GetBoundsInScreen().width();
```
[the remark this answers is not in the document]
Comment (keur) [reply browser/ui/views/speedreader/speedreader_dancing_books.cc]: fixed

File: browser/ui/views/speedreader/speedreader_dancing_books.h [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+class SpeedreaderDancingBooks : public views::View {
+ public:
+  SpeedreaderDancingBooks();
+  ~SpeedreaderDancingBooks() override;
```
[the remark this answers is not in the document]
Comment (keur) [reply browser/ui/views/speedreader/speedreader_dancing_books.h]: done

File: browser/ui/views/speedreader/speedreader_mode_bubble.cc [hunk 6]
```diff
@@ -137,6 +138,9 @@ void SpeedreaderModeBubble::Init() {
       base::BindRepeating(&SpeedreaderModeBubble::OnLinkClicked,
                           base::Unretained(this)));
   site_toggle_explanation_ = AddChildView(std::move(site_toggle_explanation));
+
+  auto speedreader_dancing_books = std::make_unique<SpeedreaderDancingBooks>();
+  AddChildView(std::move(speedreader_dancing_books));
```
[the remark this answers is not in the document]
Comment (keur) [reply browser/ui/views/speedreader/speedreader_mode_bubble.cc]: fixed

File: browser/ui/views/speedreader/speedreader_dancing_books.cc (hunk 1 above)
Comment (keur) [reply browser/ui/views/speedreader/speedreader_dancing_books.cc]: removed. just did `= default` now

**Review by simonhong — APPROVED** [review]
++

**Review by iefremov — APPROVED** [review]

**Review by karenkliu — CHANGES_REQUESTED** [review]
Hi Kevin! This looks great. Would it be possible for the graphic to sit a little closer to the bottom of the panel so it's flush? Like this:

image

It looks a little bit too floaty rn.

**Comment by keur** [comment]
@karenkliu here are the new changes

Screenshot_20210720_110310
Screenshot_20210720_110221

**Review by karenkliu — APPROVED** [review]
Looks great! 💯

**Comment by keur** [comment]
MacOS, Windows, Linux passed CI. Rebased & re-running iOS and Android.

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 10 x64` using the following build:

```
Brave | 1.29.38 Chromium: 92.0.4515.115 (Official Build) nightly (64-bit)
--- | ---
Revision | 48cb2f4029b84b003719740a6cf9ca73f374a857-refs/branch-heads/4515_105@{#4}
OS | Windows 10 OS Version 2009 (Build 19042.1110)
```

**`Example`** | **`Example`**
---------------|---------------
image | image

**Pull request merged**